crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -287,7 +297,14 @@ where
                             .find_map(|scope| scope.get(name.as_str()));
 
                         if let Some(binding_id) = binding_id {
-                            self.semantic_model.bindings[*binding_id].runtime_usage = usage;
+                            let reference_id = self.semantic_model.references.push(
+                                self.semantic_model.scope_id,
+                                stmt.range(),
+                                ReferenceContext::Runtime,
+                            );
+                            self.semantic_model.bindings[*binding_id]
```

Comment:
Same here: semantic_model.reference(binding_id, reference_id) is shorter, less to read in the future. It also hides how references are stored internally, making it easier to experiment with different representations